src/rulesets/Base/array.jl

```diff
@@ -262,19 +262,57 @@ function rrule(::typeof(hvcat), rows, values::Union{AbstractArray, Number}...)
 end
 
 #####
-##### `fill`
+##### `reverse`
 #####
 
-function rrule(::typeof(fill), value::Any, dims::Tuple{Vararg{Int}})
-    function fill_pullback(Ȳ)
-        return (NoTangent(), sum(Ȳ), NoTangent())
-    end
-    return fill(value, dims), fill_pullback
+# 1-dim case allows start/stop, N-dim case takes dims keyword, whose defaults changes in Julia 1.6.
+
+# function frule((_, xdot, _, _), ::typeof(reverse), x::AbstractArray, args...; kw...)
+#     return reverse(x, args...; kw...), reverse(xdot, args...; kw...) # seems not to work
+# end
+function frule((_, xdot, _, _), ::typeof(reverse), x::AbstractVector, start::Integer, stop::Integer)
+    return reverse(x, start, stop), reverse(xdot, start, stop)
+end
+function frule((_, xdot), ::typeof(reverse), x::AbstractArray; dims=VERSION >= v"1.6-" ? Colon() : 1)
```

Comment:
what does the `-` mean in this version bound? I couldn't find it here: http://pkgdocs.julialang.org/v1/compatibility/

We should probably observe the line limit here and in a few other places - could you please run JuliaFormatter with BlueStyle on this?

Reply:
I don't know where I got this, but:
```
julia> VERSION
v"1.7.0-beta2"

julia> VERSION >= v"1.7-"
true

julia> VERSION >= v"1.7"
false
```
However, not really necessary for 1.6 now, so will remove the `-`.